test/image/mocks/splom_multi-axis-type.json

```diff
@@ -0,0 +1,45 @@
+{
+  "data": [
+    {
+      "type": "splom",
+      "opacity": 0.9,
+      "showupperhalf": false,
+      "diagonal": {"visible": false },
+      "dimensions": [
+        {
+          "label": "numeric",
+          "values": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+        },
+        {
+          "label": "decimal",
+          "values": [-3.3, 2.2, -1.1, 0, 1.1, -2.2, 3.3, 4.4, -5, 6]
+        },
+        {
+          "label": "bool",
+          "values": [false, true, true, true, false, true, false, false, false, true]
+        },
+        {
+          "label": "0/1",
+          "values": [0, 1, 1, 1, 1, 1, 0, 0, 0, 0]
+        },
+        {
+          "label": "string",
+          "values": ["lyndon", "richard", "gerald", "jimmy", "ronald", "george", "bill", "georgeW", "barack", "donald"]
+        }
+      ]
+    }
+  ],
+  "layout": {
+    "hovermode": "closest",
+    "margin": {
+      "b": 80,
+      "l": 80,
+      "r": 30,
+      "t": 30
+    },
+    "yaxis2": {"type": "category"},
```

Comment:
This here might be tricky to find for users as e.g. nothing indicates that `yaxis2` corresponds to the boolean data array.

Perhaps we could include this in the trace `dimensions[i]` containers?

Reply:
> Perhaps we could include this in the trace `dimensions[i]` containers?

That's a great idea - `dimension.type = 'category'` to override the autotype for both axes. That would certainly reduce (though not eliminate) the chance of mismatches like https://github.com/plotly/plotly.js/pull/2899#discussion_r209777451

Reply:
haha exactly my rationale in https://github.com/plotly/plotly.js/pull/2899#discussion_r210114042 - I should read *all* your comments before making my own. Anyway, love it.